sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/PartitioningAwareFileIndex.scala

```diff
@@ -221,7 +221,16 @@ abstract class PartitioningAwareFileIndex(
         if (!fs.isDirectory(userDefinedBasePath)) {
           throw new IllegalArgumentException(s"Option '$BASE_PATH_PARAM' must be a directory")
         }
-        Set(fs.makeQualified(userDefinedBasePath))
+        def qualifiedPath(path: Path): String = fs.makeQualified(path).toString
+
+        val qualifiedBasePath = qualifiedPath(userDefinedBasePath)
+        rootPaths
+          .find(p => !qualifiedPath(p).startsWith(qualifiedBasePath))
+          .foreach { rp =>
+            throw new IllegalArgumentException(
+              s"Wrong basePath $userDefinedBasePath for the root path: $rp")
+          }
+        Set(new Path(qualifiedBasePath))
```

Comment:
we should reduce overhead as possible as we can
```
val qualifiedBasePath = fs.makeQualified(userDefinedBasePath)
val qualifiedBasePathStr = qualifiedBasePath.toString
rootPaths.find...
Set(qualifiedBasePath)
```

Reply:
Ok, I see.